samples/04.0-file-transfer/integration-tests/src/test/java/org/eclipse/dataspaceconnector/extension/sample/test/FileTransferSampleTest.java

```diff
@@ -0,0 +1,241 @@
+                            .get(LOOK_UP_CONTRACT_AGREEMENT_URI, contractNegotiationId)
+                        .then()
+                            .statusCode(HttpStatus.SC_OK)
+                            .body("state", equalTo("CONFIRMED"))
+                            .body("contractAgreementId", not(emptyString()))
+                            .extract().body().jsonPath().getString("contractAgreementId");
+
+                    localContractAgreementId.set(result);
+
+                }
+        );
+
+        contractAgreementId = localContractAgreementId.get();
+    }
+
+    /**
+     * Assert that a POST request to initiate transfer process is successful.
+     * This methods corresponds to the command in the sample: curl -X POST -H "Content-Type: application/json" -H "X-Api-Key: password" -d @samples/04.0-file-transfer/filetransfer.json "http://localhost:9192/api/v1/data/transferprocess"
+     * @throws IOException Thrown if there was an error accessing the transfer request file defined in {@link FileTransferSampleTest#TRANSFER_FILE_PATH}.
+     */
+    void assertRequestFile() throws IOException {
+        File transferJsonFile = getFileFromRelativePath(TRANSFER_FILE_PATH);
+        DataRequest sampleDataRequest = readAndUpdateTransferJsonFile(transferJsonFile, contractAgreementId);
+
+        JsonPath jsonPath = RestAssured
+            .given()
+                .headers(API_KEY_HEADER_KEY, API_KEY_HEADER_VALUE)
+                .contentType(ContentType.JSON)
+                .body(sampleDataRequest)
+            .when()
+                .post(INITIATE_TRANSFER_PROCESS_URI)
+            .then()
+                .statusCode(HttpStatus.SC_OK)
+                .body("id", not(emptyString()))
+                .extract()
+                .jsonPath();
+
+        String transferProcessId = jsonPath.get("id");
+
+        assertThat(transferProcessId).isNotEmpty();
+    }
+
+    /**
+     * Reads a transfer request file with changed value for contract agreement ID and file destination path.
+     * @param transferJsonFile A {@link File} instance pointing to a JSON transfer request file.
+     * @param contractAgreementId This string containing a UUID will be used as value for the contract agreement ID.
+     * @return An instance of {@link DataRequest} with changed values for contract agreement ID and file destination path.
+     * @throws IOException Thrown if there was an error accessing the file given in transferJsonFile.
+     */
+    DataRequest readAndUpdateTransferJsonFile(File transferJsonFile, String contractAgreementId) throws IOException {
+        // create object mapper instance
+        ObjectMapper mapper = new ObjectMapper();
+
+        // convert JSON file to map
+        DataRequest sampleDataRequest = mapper.readValue(transferJsonFile, DataRequest.class);
+
+        DataAddress newDataDestination = sampleDataRequest.getDataDestination().toBuilder().property("path", destinationFile.getAbsolutePath()).build();
```

Comment:
This could be done as follows, which would eliminate the need to introduce `toBuilder`.

```java
var dataRequest = mapper.readTree(transferJsonFile);
var parentNode = dataRequest.get("dataDestination").get("properties");
((ObjectNode) parentNode).put("path", destinationFile.getAbsolutePath());
var newDataDestination = mapper.treeToValue(dataRequest, DataRequest.class);
```